test/integration/targets/sns_topic/tasks/main.yml

```diff
@@ -116,14 +105,76 @@
         name: "{{ sns_topic_topic_name }}"
         display_name: "My new topic name"
         policy: "{{ lookup('template', 'updated-policy.json') }}"
-        <<: *aws_connection_info
       register: sns_topic_update_policy
 
     - name: assert that updating policy worked
       assert:
         that:
           - sns_topic_update_policy.changed
 
+    - name: add delivery policy
+      sns_topic:
+        name: "{{ sns_topic_topic_name }}"
+        display_name: "My new topic name"
+        delivery_policy:
+            http:
+              defaultHealthyRetryPolicy:
+                minDelayTarget: 20
+                maxDelayTarget: 20
+                numRetries: 3
+                numMaxDelayRetries: 0
+                numNoDelayRetries: 0
+                numMinDelayRetries: 0
+                backoffFunction: 'linear'
+      register: sns_topic_add_delivery_policy
+
+    - name: assert that adding delivery policy worked
+      assert:
+        that:
+          - sns_topic_add_delivery_policy.changed
+
+    - name: rerun same delivery policy
+      sns_topic:
+        name: "{{ sns_topic_topic_name }}"
+        display_name: "My new topic name"
+        delivery_policy:
+            http:
+              defaultHealthyRetryPolicy:
+                minDelayTarget: 20
+                maxDelayTarget: 20
+                numRetries: 3
+                numMaxDelayRetries: 0
+                numNoDelayRetries: 0
+                numMinDelayRetries: 0
+                backoffFunction: 'linear'
+      register: sns_topic_rerun_delivery_policy
+
+    - name: assert that rerunning delivery_policy had no effect
+      assert:
+        that:
+          - not sns_topic_rerun_delivery_policy.changed
```

Comment:
Could you please assert the values of minDelayTarget, maxDelayTarget, and numRetries, so that the changed values can also be asserted at ln176?

Reply:
Annoyingly the policies are returned as strings so I've used "| from_json" to parse it.